src/Akka.Persistence.MongoDb/Journal/MongoDbJournal.cs

```diff
@@ -288,41 +335,52 @@ protected override async Task<IImmutableList<Exception>> WriteMessagesAsync(IEnu
+
             if (HasNewEventSubscribers)
                 NotifyNewEventAppended();
             return result;
         }
+
         private async ValueTask InsertEntries(IEnumerable<JournalEntry> entries)
         {
-            if (_settings.Transaction)
+            var (perCallCts, unitedCts) = CreatePerCalLCts();
+            using (perCallCts)
+            using (unitedCts)
             {
-                var sessionOptions = new ClientSessionOptions { };
-                using (var session = await _journalCollection.Value.Database.Client.StartSessionAsync(sessionOptions/*, cancellationToken*/))
+                if (_settings.Transaction)
                 {
-                    // Begin transaction
-                    session.StartTransaction();
-                    try
+                    var sessionOptions = new ClientSessionOptions { };
+                    using (var session =
+                           await _journalCollection.Value.Database.Client.StartSessionAsync(
+                               sessionOptions, unitedCts.Token))
                     {
-                        //https://www.mongodb.com/community/forums/t/insertone-vs-insertmany-is-one-preferred-over-the-other/135982/2
-                        //https://www.mongodb.com/docs/manual/core/transactions-production-consideration/#runtime-limit
-                        //https://www.mongodb.com/docs/manual/core/transactions-production-consideration/#oplog-size-limit
-                        //https://www.mongodb.com/docs/manual/reference/limits/#mongodb-limits-and-thresholds
-                        //16MB: if is bigger than this that means you do it one by one. LET'S TALK ABOUT THIS
-                        await _journalCollection.Value.InsertManyAsync(session, entries, new InsertManyOptions { IsOrdered = true });
-                    }
-                    catch (Exception ex) 
-                    {
-                        await session.AbortTransactionAsync();
-                        throw ex;   
-                    }
+                        // Begin transaction
+                        session.StartTransaction();
+                        try
+                        {
+                            //https://www.mongodb.com/community/forums/t/insertone-vs-insertmany-is-one-preferred-over-the-other/135982/2
+                            //https://www.mongodb.com/docs/manual/core/transactions-production-consideration/#runtime-limit
+                            //https://www.mongodb.com/docs/manual/core/transactions-production-consideration/#oplog-size-limit
+                            //https://www.mongodb.com/docs/manual/reference/limits/#mongodb-limits-and-thresholds
+                            //16MB: if is bigger than this that means you do it one by one. LET'S TALK ABOUT THIS
+                            await _journalCollection.Value.InsertManyAsync(session, entries,
+                                new InsertManyOptions { IsOrdered = true });
```

Comment:
In some areas I used the CTS on the operation AND on the transaction commit - in others I only did it on the commit (as is the case here.) Should I do just use the CTS everywhere or only on the transaction commits? cc @Arkatufus

Reply:
I think it should be used everywhere, except for `AbortTransactionAsync`, that might actually need its own cancellation token, just in case the exception was caused by a time out in the first place.

Reply:
Done.